src/utils/matchMedia.test.ts

```diff
@@ -0,0 +1,61 @@
+import { addListener, removeListener } from './matchMedia'
+
+describe('addListener', () => {
+  test('adds event listener to media query', () => {
+    const mediaQuery = {
+      addEventListener: jest.fn(),
+      addListener: jest.fn(),
+    }
+
+    const listener = jest.fn()
+    addListener(mediaQuery as unknown as MediaQueryList, listener)
+
+    expect(mediaQuery.addEventListener).toHaveBeenCalledWith('change', listener)
+    expect(mediaQuery.addListener).not.toHaveBeenCalled()
+  })
+
+  test('falls back to addListener for older browsers', () => {
+    const mediaQuery = {
+      addEventListener: jest.fn(() => {
+        throw new Error('not supported')
+      }),
+      addListener: jest.fn(),
+    }
+
+    const listener = jest.fn()
+    addListener(mediaQuery as unknown as MediaQueryList, listener)
+
+    expect(mediaQuery.addEventListener).toHaveBeenCalledWith('change', listener)
+    expect(mediaQuery.addListener).toHaveBeenCalledWith(listener)
+  })
+})
+
+describe('removeListener', () => {
+  test('removes event listener from media query', () => {
+    const mediaQuery = {
```

Comment:
Type this at the declaration instead of coercing it later. This is safer in case you expand the test in the future.